### PR TITLE
Add metrics module

### PR DESCRIPTION
**Pull request by AakashKumarNain**
Added `metrics` module to add metrics that conform to the `tf.keras API`.

### CODE REVIEW

**Comment by facaiy** [comment]
Looks good, I'll also ask @seanpmorgan or @WindQAQ if they have any concerns about this change? Then let's merge it :-)

**Comment by AakashKumarNain** [comment]
> Many thanks!

No worries

**Comment by WindQAQ** [comment]
LSTM! Remember to merge the latest commit so that the builds could pass.

File: tensorflow_addons/utils/keras_utils.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 # TODO: find public API alternative to these
 from tensorflow.python.keras.losses import LossFunctionWrapper  # pylint: disable=unused-import
+from tensorflow.python.keras.metrics import Metric  # pylint: disable=unused-import
```
Comment (seanpmorgan) [inline tensorflow_addons/utils/keras_utils.py]: What was the issue with using the public `tf.keras.metrics.Metric`?
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/metrics.py#L65
```
>>> import tensorflow as tf
>>> tf.keras.metrics.Metric
<class 'tensorflow.python.keras.metrics.Metric'>
```

Comment (facaiy) [reply tensorflow_addons/utils/keras_utils.py]: Sean, I believe you're right.

File: tensorflow_addons/metrics/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+|:----------------------- |:-------------------|:---------------|
+|  |  |            |
+
+
+## Contribution Guidelines
+#### Standard API
+In order to conform with the current API standard, all metrics
+must:
+ * Be a `tf.function`.
+ * Have the signature `fn(input, axis=-1, name=None)`.
```
Comment (facaiy) [inline tensorflow_addons/metrics/README.md]: Metric must be subclass of `tf.metrics.Metric`, as Sean pointed. I think we should replace the two lines by: Inherit from `tf.metrics.Metric`.

File: tensorflow_addons/metrics/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * [Register as a keras global object](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/utils/python/keras_utils.py)
+  so it can be serialized properly.
+ * Add the addon to the `py_library` in this sub-package's BUILD file.
+
+#### Testing Requirements
+ * Simple unittests that demonstrate the layer is behaving as expected.
```
Comment (facaiy) [inline tensorflow_addons/metrics/README.md]:
```suggestion
 * Simple unittests that demonstrate the metric is behaving as expected.
```

File: tensorflow_addons/metrics/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+| Submodule | Activation  | Reference                               |
+|:----------------------- |:-------------------|:---------------|
+|  |  |            |
+
+
+## Contribution Guidelines
+#### Standard API
+In order to conform with the current API standard, all metrics
+must:
+ * Be a `tf.function`.
```
Comment (facaiy) [inline tensorflow_addons/metrics/README.md]: Metric must be subclass of `tf.metrics.Metric`, as Sean pointed. I think we should replace the two lines by: Inherit from `tf.metrics.Metric`.

```suggestion
 * Inherit from `tf.metrics.Metric`.
```

File: tensorflow_addons/metrics/README.md (hunk 2 above)
Comment (facaiy) [inline tensorflow_addons/metrics/README.md]:
```suggestion
```

File: tensorflow_addons/metrics/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ * Simple unittests that demonstrate the layer is behaving as expected.
+ * When applicable, run all unittests with TensorFlow's
+   `@run_in_graph_and_eager_modes` (for test method)
+   or `run_all_in_graph_and_eager_modes` (for TestCase subclass)
+   decorator.
+ * Add a `py_test` to this sub-package's BUILD file.
+
+#### Documentation Requirements
+ * Update the table of contents in this sub-package's README.
+
```
Comment (facaiy) [inline tensorflow_addons/metrics/README.md]:
```suggestion
```

File: tensorflow_addons/utils/keras_utils.py (hunk 1 above)
Comment (facaiy) [reply tensorflow_addons/utils/keras_utils.py]:
```suggestion
```

Comment (AakashKumarNain) [reply tensorflow_addons/utils/keras_utils.py]: I removed this

File: tensorflow_addons/metrics/README.md (hunk 3 above)
Comment (AakashKumarNain) [reply tensorflow_addons/metrics/README.md]: Updated

File: tensorflow_addons/metrics/README.md (hunk 4 above)
Comment (AakashKumarNain) [reply tensorflow_addons/metrics/README.md]: Done

File: tensorflow_addons/metrics/README.md (hunk 2 above)
Comment (AakashKumarNain) [reply tensorflow_addons/metrics/README.md]: Done

File: tensorflow_addons/metrics/README.md (hunk 5 above)
Comment (AakashKumarNain) [reply tensorflow_addons/metrics/README.md]: Removed

**Comment by facaiy** [comment]
Thank you, Aakash.

@seanpmorgan Sean, can you take an another look?

**Comment by AakashKumarNain** [comment]
Perfect. You are welcome
